### PR TITLE
Doc `std::thread::park_timeout`

### PR DESCRIPTION
**Pull request by dns2utf8**
r? @steveklabnik

### CODE REVIEW

File: src/libstd/thread/mod.rs [hunk 1]
```diff
@@ -447,6 +447,8 @@ pub fn park() {
     *guard = false;
 }
 
+/// Use [park_timeout][park_timeout].
```
Comment (steveklabnik) [inline src/libstd/thread/mod.rs]: you don't need the second `[]`s here

File: src/libstd/thread/mod.rs [hunk 2]
```diff
@@ -478,6 +483,25 @@ pub fn park_timeout_ms(ms: u32) {
...
+/// ```rust,no_run
+/// use std::thread::park_timeout;
+/// use std::time::{ Instant, Duration };
+///
+/// let timeout = Duration::from_secs(2);
+/// let beginning_park = Instant::now();
+/// park_timeout(timeout);
+///
+/// while beginning_park.elapsed() < timeout {
+///   println!("restarting park_timeout after {:?}", beginning_park.elapsed());
```
Comment (steveklabnik) [inline src/libstd/thread/mod.rs]: four space indents please!

File: src/libstd/thread/mod.rs [hunk 3]
```diff
@@ -478,6 +483,25 @@ pub fn park_timeout_ms(ms: u32) {
...
 /// Platforms which do not support nanosecond precision for sleeping will have
 /// `dur` rounded up to the nearest granularity of time they can sleep for.
+///
+/// # Example
+///
+/// Waiting for the timeout complete timeout:
+///
+/// ```rust,no_run
+/// use std::thread::park_timeout;
+/// use std::time::{ Instant, Duration };
```
Comment (steveklabnik) [inline src/libstd/thread/mod.rs]: could this be

    /// use std::time::{Instant, Duration};

**Comment by steveklabnik** [comment]
Thanks so much! I have some extremely minor nitpics :smile:

File: src/libstd/thread/mod.rs (hunk 2 above)
Comment (dns2utf8) [inline src/libstd/thread/mod.rs]: fixed

File: src/libstd/thread/mod.rs (hunk 1 above)
Comment (dns2utf8) [inline src/libstd/thread/mod.rs]: done

File: src/libstd/thread/mod.rs (hunk 3 above)
Comment (dns2utf8) [inline src/libstd/thread/mod.rs]: yes

**Comment by steveklabnik** [comment]
Great! Thanks :smile: 

@bors: r+ rollup

**Comment by dns2utf8** [comment]
Was fun :smile: Had this problem myself, so I documented it.

**Pull request merged**